storage/store.go

```diff
@@ -621,6 +621,37 @@ func (s *Store) SplitRange(origRng, newRng *Range) error {
 	return nil
 }
 
+// MergeRange expands the subsuming range to absorb the subsumed range.
+// This merge operation will fail if the two ranges are not colocated
+// on the same store.
```

Comment:
You mention this will fail if not colocated on same store. Why not expand this guarantee to say that all replicas must be colocated on the same stores. This seems like a good place to additionally check that colocation of all replicas still holds (see previous comment).